### DISCUSSION TITLE
How to document a response of type StreamingResponse in OpenAPI?

### DISCUSSION THREAD

**Discussion by jesumyip** [fastapi/fastapi#3881] [Questions] [answered]
Hi everyone.

My FastAPI endpoint returns a StreamingResponse of text/plain. This API allows a user to download a file. The API definition looks like this.

@router.get(f"/{api_name}",
    tags=[f"{api_name}"],
    summary = "Bla bla bla.",
    response_description="Streamed content for file download.",
    responses = {
        **res.responses,
        200 : { "description" : "Binary stream of file.", "content": "text/plain" }
    }

which fails. The automatic redoc documentation shows this only. 
![image](https://user-images.githubusercontent.com/87511866/132708027-884e7cab-142b-48c3-8825-d48cef3a8d6a.png)

Is there a way to achieve this? I am expecting the automatic documentation to show a response sample of 200 that shows content type of text/plain.

**Comment by jesumyip** [comment]
I ended up "cheating" by using https://fastapi.tiangolo.com/advanced/extending-openapi/#override-the-method

**Comment by ghandic** [comment, marked as answer]
@jesumyip FastAPI utilises Starlette under the hood so you can add national parameters to the `StreamingResponse` object to achieve what you want

https://www.starlette.io/responses/#streamingresponse

**Reply by jesumyip** [reply]
Thanks!

**Comment by hadrien** [comment]
Stumbled upon this as I had the same issue rendering relevant doc with a streamed response. 

Here is what I did:

```python
@app.get(
    "/example",
    responses={
        200: {
            "content": {"text/plain": {}},
            "description": "Stream plain text using utf8 charset.",
        }
    },
    response_class=StreamingResponse,
)
async def example():
    pass
```

How it renders:

<img width="1425" alt="image" src="https://github.com/tiangolo/fastapi/assets/75603/ec6f707b-9c23-44c5-a67d-bb6f283fccca">
